### PR TITLE
Internal trackers APIs

### PR DESCRIPTION
**Pull request by eupp**
This PR implements a refactoring to introduce separate internal event trackers API:

- `MonitorTracker` tracks monitor events (lock, unlock, wait, notify)
- `ParkingTracker` tracks parking events (park, unpark)
- `ObjectTracker` tracks new objects creation and changes in object graph topology. 

Different strategies can implement these tracker interfaces for their needs. 

We need this refactoring for two features:
- new model checking strategy need to implement its own logic for tracking aforementioned events;
- to support blocking operations, stress strategy also need to track parking events.

### CODE REVIEW

**Comment by eupp** [comment]
@avpotapov00 could you please have a look at this PR and check if it aligns with your vision for the parking tracker for stress strategy.

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ParkingTracker.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * Tracks parking operations.
+ */
+interface ParkingTracker {
+
+    /**
+     * Parks the specified thread.
+     *
+     * @param iThread the id of the thread to be parked
+     */
+    fun park(iThread: Int)
```
Comment (ndkoval) [inline src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ParkingTracker.kt:23]: iThread -> threadId

**Comment by ndkoval** [comment]
@eupp, great, I will take a look at the PR. Meanwhile, could you please investigate why the build on Windows has failed?

**Comment by eupp** [comment]
> Meanwhile, could you please investigate why the build on Windows has failed?

Looks like unrelated flakiness problem. I've re-run the build and it is green now.

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ParkingTracker.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+     * @param threadId the id of the thread to be parked
+     */
+    fun park(threadId: Int)
+
+    /**
+     * Waits for the specified thread to be unparked.
+     *
+     * @param threadId the id of the thread to check for parking status.
+     * @return true if the thread should continue waiting,
+     *   false if it was already unparked.
```
Comment (ndkoval) [inline src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ParkingTracker.kt:30]: Should it return `false` if the execution has finished?

**Review by ndkoval — APPROVED** [review]

**Comment by ndkoval** [comment]
Hi @eupp, I've approved the PR. Thanks for extracting the logic and improving the code structure! 

I've left a minor comment -- please address it and merge the PR.

File: src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ParkingTracker.kt (hunk 2 above)
Comment (eupp) [reply src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ParkingTracker.kt:30]: So can I merge this PR? @ndkoval

**Pull request merged**